elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractor.java

```diff
@@ -62,7 +64,24 @@ public JoinExpressionExtractor(Context context, ColumnProjection column) {
          * For the scenario: col1:{{col2}}, col2:{{col3}}, col3:{{join.col1}}
          * Creating new visitor with correct ColumnProjection, so that default arguments can be extracted correctly.
          */
-        JoinExpressionExtractor visitor = new JoinExpressionExtractor(this.context, reference.getColumn());
+        ColumnProjection column = reference.getColumn();
+
+        Map<String, Argument> fixedArgs = reference.getFixedArguments();
```

Comment:
Rather than have a separate set of fixed arguments here,  I would expect the logical reference column to already have these arguments merged in.

Reply:
fixed arguments must override calling column's arguments so keeping them separately